### ISSUE TITLE
Display useful error message with Exception DockerRegistry2::NotFound

### ISSUE THREAD

**Issue by honeyankit** [deitch/docker_registry2#67]
## Context
When an image is not found in the private registry, it throws an exception of DockerRegistry2::NotFound and lot of stack trace but this info is not useful for the end user. [[code](https://github.com/deitch/docker_registry2/blob/970b4aafd1259e4867293f9a6be180a46730b366/lib/registry/registry.rb#L284-L285)]

Error:
```
updater | ERROR <job_500122728> Error processing myreg/ubuntu (DockerRegistry2::NotFound)
updater | ERROR <job_500122728> 404 Not Found
updater | ERROR <job_500122728> /home/dependabot/dependabot-updater/vendor/ruby/3.1.0/gems/docker_registry2-1.12.0/lib/registry/registry.rb:285:in `rescue in doreq'
updater | ERROR <job_500122728> /home/dependabot/dependabot-updater/vendor/ruby/3.1.0/gems/docker_registry2-1.12.0/lib/registry/registry.rb:269:in `doreq'
updater | ERROR <job_500122728> /home/dependabot/dependabot-updater/vendor/ruby/3.1.0/gems/docker_registry2-1.12.0/lib/registry/registry.rb:26:in `doget'
```

## Proposed Solution

A useful error message indicating the "image that is attempting to be retrieved does not exist in the registry" should be displayed when the "DockerRegistry2::NotFound" exception is thrown.

**Comment by jeffwidman** [comment]
@deitch are you open to a PR on this?

**Comment by deitch** [comment]
Definitely.

**Comment by honeyankit** [comment]
@deitch : Any update on this issue?

**Comment by deitch** [comment]
@jeffwidman @honeyankit I saw the linked dependabot PR. I would be happy to see a PR here so you won't have that issue there.

**Comment by jeffwidman** [comment]
Can this be closed now that #68 is merged?
